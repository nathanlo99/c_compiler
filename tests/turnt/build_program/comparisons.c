
int wain(int a, int b) {
  if (a < b) {
    println(a);
  }
  // NOTE: This is equivalent to a < b and should be detected as such by GVN
  if (b > a) {
    println(b);
  }
  return a;
}
