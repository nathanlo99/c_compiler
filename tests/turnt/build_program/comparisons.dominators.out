Function: wain
  Block: wainEntry
  - Immediate dominator: (none)
  - Dominance frontier: {}
  Block: ifTrue0
  - Immediate dominator: wainEntry
  - Dominance frontier: {ifFalse0}
  Block: ifFalse0
  - Immediate dominator: wainEntry
  - Dominance frontier: {}
  Block: ifTrue1
  - Immediate dominator: ifFalse0
  - Dominance frontier: {ifFalse1}
  Block: ifFalse1
  - Immediate dominator: ifFalse0
  - Dominance frontier: {}
